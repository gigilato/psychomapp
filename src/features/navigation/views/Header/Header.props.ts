import { ReactNode } from 'react'

export interface IHeaderProps {
  height?: number
  children?: ReactNode
}
