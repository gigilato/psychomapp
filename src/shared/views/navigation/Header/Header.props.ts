import { ReactNode } from 'react'

import { IBoxProps } from '$atoms'

export interface IHeaderProps extends IBoxProps {
  height?: number
  children?: ReactNode
}
