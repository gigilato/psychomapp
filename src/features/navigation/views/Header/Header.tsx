import { Canvas, Skia, Path } from '@shopify/react-native-skia'
import { useWindowDimensions } from 'react-native'

import { Box } from '$atoms'
import { IHeaderProps } from '$features/navigation/views/Header/Header.props'
import { colors } from '$theme'

export const Header = ({ height = 200, children }: IHeaderProps) => {
  const { width } = useWindowDimensions()
  const style = { width, height }

  const path = Skia.Path.Make()
  path.moveTo(0, 0)
  path.lineTo(0, height * 0.75)
  path.cubicTo(width * 0.25, height * 0.9, width / 2, height, width, height * 0.95)
  path.lineTo(width, 0)
  path.close()

  return (
    <Box style={style}>
      <Canvas style={style}>
        <Path path={path} strokeWidth={2} style="fill" color={colors.secondary.classic} />
      </Canvas>
      <Box absoluteFill>{children}</Box>
    </Box>
  )
}
