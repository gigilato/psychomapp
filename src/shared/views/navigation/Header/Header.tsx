import { Canvas, Skia, Path } from '@shopify/react-native-skia'
import { useWindowDimensions } from 'react-native'

import { Box } from '$atoms'
import { colors } from '$theme'

import { IHeaderProps } from './Header.props'

export const Header = ({ height = 200, children, ...props }: IHeaderProps) => {
  const { width } = useWindowDimensions()
  const style = { width, height }

  const path = Skia.Path.Make()
  path.moveTo(0, 0)
  path.lineTo(0, height * 0.75)
  path.cubicTo(width * 0.25, height * 0.9, width / 2, height, width, height * 0.95)
  path.lineTo(width, 0)
  path.close()

  return (
    <Box style={style} className="bg-white-classic">
      <Canvas style={style}>
        <Path path={path} strokeWidth={2} style="fill" color={colors.secondary.classic} />
      </Canvas>
      <Box absoluteFill {...props}>
        {children}
      </Box>
    </Box>
  )
}
