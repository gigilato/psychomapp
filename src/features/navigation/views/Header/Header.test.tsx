import { render } from '@testing-library/react-native'
import React from 'react'

import { Center, Text } from '$atoms'

import { Header } from './Header'

describe('Header', () => {
  it('should renders correctly', () => {
    const { toJSON } = render(<Header />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should renders with children', () => {
    const { toJSON } = render(
      <Header>
        <Center className="flex-1">
          <Text>Header</Text>
        </Center>
      </Header>
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
