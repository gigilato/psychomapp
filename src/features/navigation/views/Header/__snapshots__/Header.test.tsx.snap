// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should renders correctly 1`] = `
<View
  style={
    [
      [
        undefined,
        {
          "height": 200,
          "width": 750,
        },
      ],
    ]
  }
>
  <View
    style={
      {
        "height": 200,
        "width": 750,
      }
    }
  >
    <View
      color="#BB99DD"
      path={
        {
          "addCircle": [MockFunction],
          "close": [MockFunction] {
            "calls": [
              [],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "cubicTo": [MockFunction] {
            "calls": [
              [
                187.5,
                180,
                375,
                200,
                750,
                190,
              ],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "lineTo": [MockFunction] {
            "calls": [
              [
                0,
                150,
              ],
              [
                750,
                0,
              ],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
              {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "moveTo": [MockFunction] {
            "calls": [
              [
                0,
                0,
              ],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
            ],
          },
        }
      }
      strokeWidth={2}
      style="fill"
    />
  </View>
  <View
    style={
      [
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ],
      ]
    }
  />
</View>
`;

exports[`Header should renders with children 1`] = `
<View
  style={
    [
      [
        undefined,
        {
          "height": 200,
          "width": 750,
        },
      ],
    ]
  }
>
  <View
    style={
      {
        "height": 200,
        "width": 750,
      }
    }
  >
    <View
      color="#BB99DD"
      path={
        {
          "addCircle": [MockFunction],
          "close": [MockFunction] {
            "calls": [
              [],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "cubicTo": [MockFunction] {
            "calls": [
              [
                187.5,
                180,
                375,
                200,
                750,
                190,
              ],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "lineTo": [MockFunction] {
            "calls": [
              [
                0,
                150,
              ],
              [
                750,
                0,
              ],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
              {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "moveTo": [MockFunction] {
            "calls": [
              [
                0,
                0,
              ],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
            ],
          },
        }
      }
      strokeWidth={2}
      style="fill"
    />
  </View>
  <View
    style={
      [
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        [
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          [
            undefined,
            undefined,
          ],
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "SpaceGrotesk-Regular",
            },
            {
              "fontSize": 14,
            },
            {
              "lineHeight": 18,
            },
          ]
        }
      >
        Header
      </Text>
    </View>
  </View>
</View>
`;
